specs/GetFavicon.md

```diff
@@ -0,0 +1,172 @@
+                        {
+                            CHECK_FAILURE(errorCode);
+                            Gdiplus::Bitmap iconBitmap(iconStream);
+                            wil::unique_hicon icon;
+                            if (!iconBitmap.GetHICON(&icon))
+                            {
+                                m_favicon = std::move(icon);
+                                SendMessage(
+                                    m_appWindow->GetMainWindow(), WM_SETICON,
+                                    ICON_SMALL, (LPARAM)m_favicon.get());
+                                m_statusBar.Show(strUrl);
+                            }
+
+                            return S_OK;
+                        })
+                        .Get());
+            }
+            return S_OK;
+        }).Get(), &m_faviconChangedToken));
+}
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    IRandomAccessStream stream = await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+Note that even if a web page does not have a Favicon, the FaviconChanged event
+is raised when the page is navigated to. The Favicon would be an
+empty image stream and empty Uri for the lack of a favicon. The end developer is expected to handle this scenario.
+Otherwise, we raise the FaviconChanged with an observed change to the
+Favicon. In that scenario, the CoreWebView2 has an updated value for the FaviconUri property, and the GetFavicon method to match the updated favicon.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// This interface is a handler for when the `Favicon` is changed.
+/// The sender is the ICoreWebView2 object the top-level document of 
+/// which has changed favicon and the eventArgs is nullptr. Use the 
+/// FaviconUri property and GetFavicon method to obtain the favicon 
+/// data. The second argument is always null.
+/// For more information see `add_FaviconChanged`.
+[uuid(2913DA94-833D-4DE0-8DCA-900FC524A1A4), object, pointer_default(unique)]
+interface ICoreWebView2FaviconChangedEventHandler : IUnknown {
+  /// Called to notify the favicon changed.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+/// This interface is a handler for the completion of the population of
+/// `imageStream`.
+/// `errorCode` returns S_OK if the API succeeded.
+/// The image is returned in the `faviconStream` object. If there is no image
+/// then no data would be copied into the imageStream.
+/// For more details, see the `GetFavicon` API.
+[uuid(A2508329-7DA8-49D7-8C05-FA125E4AEE8D), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalGetFaviconCompletedHandler : IUnknown {
```

Comment:
not sure if we want to use "experimental" in the API name at this point.  We appear to have omitted this word in the parameter in "HRESULT GetFavicon" below.